dictcc.el

```diff
@@ -271,11 +302,18 @@ At the moment they are of the form `<tr id='trXXX'></tr>'."
                                (dictcc--insert-candidate #'cdr))))))
     (helm :prompt "Filter: " :sources (list source))))
 
+(defun dictcc--select-sourcelang ()
+  "Select source language with completing read."
+  (let ((lang (completing-read "Languages: "  dictcc-source-languages-alist)))
+    (cdr (assoc lang dictcc-source-languages-alist))))
+
 ;;;###autoload
 (defun dictcc (query)
   "Search dict.cc for QUERY and insert a result at point."
   (interactive "sQuery: \n")
-  (dictcc--request query))
+  (let* ((asklang current-prefix-arg)
+	 (dictcc-source-lang (if asklang (dictcc--select-sourcelang) dictcc-source-lang)))
```

Comment:
Usual lisp formatting has opening parens aligned with other parens on the same level. Meaning that this line should be indented one more space. If you use emacs' built-in indentation, you should get all of these correct automatically. The is true further up at the definition of `dictcc-source-languages-alist` and `dictcc--request-url` after you removed the `asklang` parameter again

Reply:
`C-x h TAB` will also re-indent the whole buffer correctly.

Reply:
I used automatic indentation and otherwise tried to stick to the formatting of your code. I hope that I have got it right this time.

Reply:
I just found that I had indent-tabs mode activated. This might explain all the hurley-burley. I used untabify to fix this.